### PR TITLE
openmoji-color,openmoji-black: 14.0.0 -> 15.0.0

### PR DESCRIPTION
**Pull request by 999eagle**
## Description of changes

Release Notes: https://github.com/hfg-gmuend/openmoji/releases/tag/15.0.0

The build system has changed completely between 14.0.0 and 15.0.0 and generating all font formats upstream builds takes a *very* long time. I've experimented with the available fonts and `colr1_svg` (built from `glyf_colr_1` using `maximum_color`) seems to have the best support across font renderers, so that's what I've chosen for `openmoji-color` by default.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 999eagle** [comment]
@fgaz ping

File: pkgs/data/fonts/openmoji/default.nix [hunk 1]
```diff
@@ -1,93 +1,81 @@
...
-    # this is API change in glob >9
-    substituteInPlace helpers/generate-font-glyphs.js \
-      --replace "require('glob').sync" "require('glob').globSync"
+  methods_black = builtins.filter (m: builtins.elem m fontFormats) methods.black;
+  methods_color = builtins.filter (m: builtins.elem m fontFormats) methods.color;
+  saturations = lib.optional (methods_black != [ ]) "black" ++ lib.optional (methods_color != [ ]) "color";
+  maximumColorVersions = lib.optionals (buildMaximumColorFonts != false) (
+    lib.optional (builtins.elem "glyf_colr_0" fontFormats) "0"
+    ++ lib.optional (builtins.elem "glyf_colr_1" fontFormats) "1"
+  );
```
Comment (fgaz) [inline pkgs/data/fonts/openmoji/default.nix:51]: These should go under the `env` attribute

Comment (fgaz) [reply pkgs/data/fonts/openmoji/default.nix:51]: or outside of the callPackage attrset in a let block if they are not used directly in the script

Comment (999eagle) [reply pkgs/data/fonts/openmoji/default.nix]: Done

**Review by fgaz — COMMENTED** [review]
Please change the commit to `openmoji-black,openmoji-color: 14.0.0 -> 15.0.0` so ofborg can recognize it

File: pkgs/data/fonts/openmoji/default.nix [hunk 2]
```diff
@@ -1,93 +1,83 @@
...
-      --replace "require('glob').sync" "require('glob').globSync"
+  env = rec {
+    methods_black = builtins.filter (m: builtins.elem m fontFormats) methods.black;
+    methods_color = builtins.filter (m: builtins.elem m fontFormats) methods.color;
+    saturations = lib.optional (methods_black != [ ]) "black" ++ lib.optional (methods_color != [ ]) "color";
+    maximumColorVersions = lib.optionals (buildMaximumColorFonts != false) (
+      lib.optional (builtins.elem "glyf_colr_0" fontFormats) "0"
+      ++ lib.optional (builtins.elem "glyf_colr_1" fontFormats) "1"
+    );
+  };
```
[the remark this answers is not in the document]
Comment (999eagle) [reply pkgs/data/fonts/openmoji/default.nix]: I haven't really figured out a good way to pass arrays when using `env`, so I hope this is okay? Do you know of a better way to do this?

[the remark this answers is not in the document]
Comment (fgaz) [reply pkgs/data/fonts/openmoji/default.nix]: Sorry, I should have written that (in this case at least) it should be enough to call `toString` on those

[the remark this answers is not in the document]
Comment (999eagle) [reply pkgs/data/fonts/openmoji/default.nix]: No that would only work as long as each of those lists has 0 or 1 elements. I think this was broken before as well but should be fixed now.

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 275871` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>openmoji-black</li>
    <li>openmoji-color</li>
  </ul>
</details>

old files

```
results/openmoji-black/share/fonts/truetype/OpenMoji-Black.ttf
results/openmoji-color/share/fonts/truetype/OpenMoji-Color.ttf
```

new file:
```
results/openmoji-black/share/fonts/truetype/OpenMoji-black-glyf.ttf
results/openmoji-black/share/fonts/woff2/OpenMoji-black-glyf.woff2
results/openmoji-color/share/fonts/truetype/OpenMoji-color-colr0_svg.ttf
results/openmoji-color/share/fonts/woff2/OpenMoji-color-colr0_svg.woff2
```

---

Looks pretty good to me. A concern i have is that adding a unsupported entry into `fontFormats` will be ignored and not even show a eval trace to the user. Could you add an assert ensuring all configured values are supported?

File: pkgs/data/fonts/openmoji/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (fgaz) [reply pkgs/data/fonts/openmoji/default.nix]: Sorry again, I believe I made a mistake in suggesting `env`. I had a look at the builder code and I found that top-level attributes are indeed the right way to pass structured data, while `env` is specifically for plain environment variables, so your first iteration was totally fine. I'll revert to that commit and merge in a while.

**Review by fgaz — APPROVED** [review]

**Pull request merged**